cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,285 @@
+    ```
+    The details of the resource type `Mycompany.Messaging` are displayed.
+
+7. Deb deletes a resource-type Plaid from the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/plaidService
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource type plaidResource from the resource provider Mycompany.Messaging? (yes/no)
+    ```
+    The resource-type deletion errors out as there are resources provisioned and running under the resource type. The resource myplaid should be deleted before deleting the resource type. 
+
+## Finalized set of commands
+
+### rad resource-type
+
+#### 1. rad resource-type create
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Each resource type can define multiple API versions, and each API version defines a schema that resource instances conform to. Inputs can be passed in as a file or as flags.
+
+Create a resource type from a file 
+    
+```bash
+rad resource-type create Mycompany.Messaging/plaidResource -f plaid.yaml
+```
+
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource type definition
+
+#### 2. rad resource-type list
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can list all the resource types available in the resource provider.
+
+```bash
+rad resource-type list --resource-provider Mycompany.Messaging
+NAME            TYPE                STATE
+plaidResource   Mycompany.Messaging Succeeded
+resourceType2   Mycompany.Messaging Succeeded
+```
+
+#### 3. rad resource-type show
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can display the details of the resource types.
+
+```bash
+rad resource-type show Mycompany.Messaging/plaidResource
+NAME            TYPE                STATE
+plaidResource   Mycompany.Messaging Succeeded
+```
+
+#### 4. rad resource-type delete
+Resource types are the entities that implement resource types such as 'Applications.Core/containers'. Users can delete a resource type.
+
+```bash
+rad resource-type delete Mycompany.Messaging/plaidResource
+```
+We decided to prompt the user with a warning message if there are resources provisioned and running under the resource type. The resource-type deletion errors out as there are resources provisioned and running under the resource type. The resource myplaid should be deleted before deleting the resource type. 
+
+### rad resource create
+Resources are instances of resource types such as 'Applications.Core/containers'. Inputs can be passed in as a file or as flags.
+
+Create a resource from a file 
+    
+```bash
+rad resource create -f myplaid.json
+```
+
+**Flags**
+- `-f`, `--fom-file` : The file that contains the resource properties
+
+### rad resource-provider
```

Comment:
Why does this exist? What's the use case? I thought we agreed to not surface this in the CLI UX?